packages/react-web/src/inbox/useEnhanceConversations.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  ProfileId,
+  ProfileOwnedByMe,
+  ReadResult,
+  UnspecifiedError,
+  useProfiles,
+} from '@lens-protocol/react';
+import { assertError } from '@lens-protocol/shared-kernel';
+import { useConversations } from '@xmtp/react-sdk';
+import { useEffect, useMemo, useState } from 'react';
+
+import { extractPeerProfileId, createUniqueConversationId, notEmpty } from './helpers';
+import { EnhancedConversation } from './types';
+
+export type EnhanceConversationsRequest = {
+  profile: ProfileOwnedByMe;
+};
+
+export function useEnhanceConversations(
+  useConversationsResult: ReturnType<typeof useConversations>,
+  { profile }: EnhanceConversationsRequest,
+): ReadResult<EnhancedConversation[], UnspecifiedError | Error> {
+  const { conversations, error: resultError, isLoading: resultLoading } = useConversationsResult;
+  const [enhancedConversations, setEnhancedConversations] = useState<EnhancedConversation[]>([]);
+
+  const conversationToProfileIdMap: Record<string, ProfileId | undefined> = useMemo(() => {
+    return conversations.reduce((acc, c) => {
+      const peerProfileId = extractPeerProfileId(c.context?.conversationId, profile.id);
+
+      return {
+        ...acc,
+        [createUniqueConversationId(c)]: peerProfileId,
+      };
+    }, {});
+  }, [conversations, profile.id]);
+
+  const uniqueProfileIds: ProfileId[] = useMemo(() => {
+    const ids = Object.values(conversationToProfileIdMap).filter(notEmpty);
+    return [...new Set(ids)];
+  }, [conversationToProfileIdMap]);
```

Comment:
Why not compute `conversationToProfileIdMap` and `uniqueProfileIds` in the same `useMemo` instead of chaining them?